assets/js/xrp-ledger-toml-checker.js

```diff
@@ -244,16 +255,161 @@ async function validate_address_domain_on_net(address, domain, net) {
+  const url = "wss://xrplcluster.com"
+  if (typeof socket !== "undefined" && socket.readyState < 2) {
+    socket.close()
+  }
+  const data = {
+    "command": "account_info",
+    "account": wallet,
+  }
+  socket = new WebSocket(url)
+  socket.addEventListener('message', (event) => {
+    let data;
+    try {
+      data = JSON.parse(event.data)
+      if (data.status === 'success') {
+          if (data.result.account_data.Domain) {
+            try {
+              checkingLog.resolve('SUCCESS').addClass(CLASS_GOOD)
+              decodeHexWallet(data.result.account_data.Domain)
+            } catch(e) {
+              console.log(e)
+              checkingLog.resolve('ERROR').addClass(CLASS_BAD).after('<p>Error decoding domain field: ' + data.result.account_data.Domain + '</p>')
+            }
+          } else {
+            checkingLog.resolve('ERROR').addClass(CLASS_BAD).after(TIPS_2)
+          }
+      } else {
+        checkingLog.resolve('ERROR').addClass(CLASS_BAD).after(TIPS_1)
+      }
+    } catch {
+      return false
+    }
+  })
+  socket.addEventListener('open', () => {
+    socket.send(JSON.stringify(data))
+  })
 }
 
+async function parseXRPLTomlWallet(data) {
+  let parsed
+  let logTOML = makeLogEntryWallet("Parsing TOML data...")
+  try {
+    parsed = TOML(data)
+    logTOML.resolve("SUCCESS").addClass(CLASS_GOOD)
+  } catch(e) {
+    logTOML.resolve(e).addClass(CLASS_BAD)
+    return
+  }
+
+  if (parsed.hasOwnProperty("METADATA")) {
+    const metadata_type = makeLogEntryWallet("Metadata section: ")
+    if (Array.isArray(parsed.METADATA)) {
+      metadata_type.resolve("Wrong type - should be table").addClass(CLASS_BAD)
+    } else {
+      metadata_type.resolve("Found").addClass(CLASS_GOOD)
+
+      if (parsed.METADATA.modified) {
+        const mod_log = makeLogEntryWallet("Modified date: ")
+        try {
+          mod_log.resolve(parsed.METADATA.modified.toISOString()).addClass(CLASS_GOOD)
+        } catch(e) {
+          mod_log.resolve("INVALID").addClass(CLASS_BAD)
+        }
+      }
+    }
+  }
+
+  async function listEntriesWallet(name, list, fields) {
+    let found = false;
+    let list_wrap = $("<p>"+name+"</p>")
+    let list_ol = $("<ol>").appendTo(list_wrap)
+    for (i=0; i<list.length; i++) {
+      let entry_wrap = $("<li>").appendTo(list_ol)
+      let entry_def = $("<ul class='mb-3'>").appendTo(entry_wrap)
+      let entry = list[i]
+      for (j=0; j<fields.length; j++) {
+        let fieldname = fields[j]
+        if (entry[fieldname] !== undefined) {
+            let field_def = $("<li><strong>"+fieldname+": </strong>").appendTo(entry_def)
+            $(" <span class='"+fieldname+"'>").text(entry[fieldname]).appendTo(field_def)
+        }
+      }
+      if (entry['address'] === wallet) {
+        entry_def.append('<li class="badge badge-success">Domain Validated <i class="fa fa-check-circle"></i></li>')
+        found=true;
+      }
```

Comment:
It's a confusing difference from the domain-based TOML lookup that this version shows all the listed accounts, but only checks the domain verification of one of them. To better suit the purpose of the account-based lookup, I suggest revising this section so that it _does not display_ any accounts other than the matching one.